src/Microsoft.Extensions.Configuration.FileExtensions/FileConfigurationProvider.cs

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.Extensions.FileProviders;
+
+namespace Microsoft.Extensions.Configuration
+{
+    /// <summary>
+    /// Base class for file based <see cref="ConfigurationProvider"/>.
+    /// </summary>
+    public abstract class FileConfigurationProvider : ConfigurationProvider
+    {
+        /// <summary>
+        /// Gets a value that determines if this instance of <see cref="FileConfigurationProvider"/> is optional.
+        /// </summary>
+        public bool Optional { get; set; }
```

Comment:
As once you have a ctor, all the derived classes have to have it and call base feeding in parameters..that is way more boilerplate